AutomatedTesting/Gem/PythonTests/EditorPythonTestTools/editor_python_test_tools/editor_entity_utils.py

```diff
@@ -313,6 +313,42 @@ def get_type_ids(component_names: list, entity_type: EditorEntityType = EditorEn
             bus.Broadcast, "FindComponentTypeIdsByEntityType", component_names, entity_type.value)
         return type_ids
 
+    def is_property_visible(self, component_property_path: str, expected: bool = True) -> bool:
+        """
+        Used to determine if a property is visible.
+        param component_property_path: String of component property. (e.g. 'Settings|Visible')
+        :param expected: The expected visibility (default True)
+        :return: None
+        """
+        visible_key = 1
+
+        get_component_property_outcome = self.get_property_type_visibility()
+        result = get_component_property_outcome[component_property_path][visible_key]
+
+        assert (result == "Visible") == expected, \
+            f"Failure: Property visibility was not {expected} for" \
+            f"{self.get_component_name()} : {component_property_path}"
+
+        return result
+
+    def toggle_property_switch(self, component_property_path: str) -> None:
+        """
+        Used to toggle a property switch
+        param component_property_path: String of component property. (e.g. 'Settings|Visible')
+        :return: None
+        """
+        start_value = self.get_component_property_value(component_property_path)
+        if start_value:
+            self.set_component_property_value(component_property_path, False)
+        else:
+            self.set_component_property_value(component_property_path, True)
+
+        end_value = self.get_component_property_value(component_property_path)
+
+        assert (start_value != end_value), \
+            f"Failure: Could not toggle the switch for " \
+            f"{self.get_component_name()} : {component_property_path}."
```

Comment:
`set_componenet_property_value` only validates if it was properly set by the bus, which will do so even if the previous property was the same value as before. This puts in an additional layer to validate that a toggle properly toggled.

Reply:
`set_component_property_value` is only checking for error fail it's not validating the set. any validation of properties should happen at a higher abstraction layer for testing. this class is just for abstracting the ebus calls away from python and making it object oriented so that a component can do actions.
I'd rather a test explicitly know its setting a particular value and test the actual value itself rather then put this behavior in here. you can write a test layer that toggles values and validates them.

Reply:
This has been completely refactored into a test abstraction layer.